sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveDefaultColumns.scala

```diff
@@ -0,0 +1,250 @@
+      val expanded: UnresolvedInlineTable = addMissingDefaultColumnValues(table).getOrElse(table)
+      replaceExplicitDefaultColumnValues(analyzer, expanded).getOrElse(table)
+
+    case i@InsertIntoStatement(_, _, _, project: Project, _, _) =>
+      enclosingInsert = Some(i)
+      val expanded: Project = addMissingDefaultColumnValues(project).getOrElse(project)
+      val replaced: Option[LogicalPlan] = replaceExplicitDefaultColumnValues(analyzer, expanded)
+      if (replaced.isDefined) i.copy(query = replaced.get) else i
+  }
+
+  // Helper method to check if an expression is an explicit DEFAULT column reference.
+  private def isExplicitDefaultColumn(expr: Expression): Boolean = expr match {
+    case u: UnresolvedAttribute if u.name.equalsIgnoreCase(CURRENT_DEFAULT_COLUMN_NAME) => true
+    case _ => false
+  }
+
+  /**
+   * Updates an inline table to generate missing default column values.
+   */
+  private def addMissingDefaultColumnValues(
+    table: UnresolvedInlineTable): Option[UnresolvedInlineTable] = {
+    assert(enclosingInsert.isDefined)
+    val numQueryOutputs: Int = table.rows(0).size
+    val schema = getInsertTableSchemaWithoutPartitionColumns.getOrElse(return None)
+    val newDefaultExpressions: Seq[Expression] = getDefaultExpressions(numQueryOutputs, schema)
+    val newNames: Seq[String] = schema.fields.drop(numQueryOutputs).map { _.name }
+    if (newDefaultExpressions.nonEmpty) {
+      Some(table.copy(
+        names = table.names ++ newNames,
+        rows = table.rows.map { row => row ++ newDefaultExpressions }))
+    } else {
+      None
+    }
+  }
+
+  /**
+   * Adds a new expressions to a projection to generate missing default column values.
+   */
+  private def addMissingDefaultColumnValues(project: Project): Option[Project] = {
+    val numQueryOutputs: Int = project.projectList.size
+    val schema = getInsertTableSchemaWithoutPartitionColumns.getOrElse(return None)
+    val newDefaultExpressions: Seq[Expression] = getDefaultExpressions(numQueryOutputs, schema)
+    val newAliases: Seq[NamedExpression] =
+      newDefaultExpressions.zip(schema.fields).map {
+        case (expr, field) => Alias(expr, field.name)()
+      }
+    if (newDefaultExpressions.nonEmpty) {
+      Some(project.copy(projectList = project.projectList ++ newAliases))
+    } else {
+      None
+    }
+  }
+
+  /**
+   * This is a helper for the addMissingDefaultColumnValues methods above.
+   */
+  private def getDefaultExpressions(numQueryOutputs: Int, schema: StructType): Seq[Expression] = {
+    val remainingFields: Seq[StructField] = schema.fields.drop(numQueryOutputs)
+    val numDefaultExpressionsToAdd: Int = {
+      if (SQLConf.get.useNullsForMissingDefaultColumnValues) {
+        remainingFields.size
+      } else {
+        remainingFields.takeWhile(_.metadata.contains(CURRENT_DEFAULT_COLUMN_METADATA_KEY)).size
+      }
+    }
+    Seq.fill(numDefaultExpressionsToAdd)(UnresolvedAttribute(CURRENT_DEFAULT_COLUMN_NAME))
```

Comment:
this looks a bit risky to create `UnresolvedAttribute` here. How do we guarantee that it won't be resolved as table columns?